test-suite/testkit/tests/counter/main.rs

```diff
@@ -906,3 +909,184 @@ fn test_explorer_add_invalid_transaction() {
         ApiError::BadRequest(ref body) if body == error_body
     );
 }
+
+#[test]
+fn test_explorer_api_with_before_transactions_error() {
+    let (mut testkit, api) = init_testkit();
+    let key_pair = gen_keypair();
+    let tx = key_pair.increment(SERVICE_ID, 13);
+
+    // This tx lead to error in before_transaction on the next transaction
+    testkit.create_block_with_transaction(tx.clone());
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&CallStatusQuery {
+            height: Height(1),
+            service_id: SERVICE_ID,
+        })
+        .get::<CallStatusResponse>("v1/call_status/before_transactions")
+        .expect("Explorer Api unexpectedly failed");
+    assert!(response.status.0.is_ok());
+
+    let tx = key_pair.increment(SERVICE_ID, 1);
+    // So perform one more tx to check the error
+    testkit.create_block_with_transaction(tx.clone());
+
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&CallStatusQuery {
+            height: Height(2),
+            service_id: SERVICE_ID,
+        })
+        .get::<CallStatusResponse>("v1/call_status/before_transactions")
+        .expect("Explorer Api unexpectedly failed");
+    assert!(response.status.0.is_err());
```

Comment:
Nit: This is redundant given that `unwrap_err()` is called later. Also, a separate `let err = response.status.0.unwrap_err()` binding would probably decrease the number of lines 🙃